src/lib/containers/drug_upload_tool/UserFileGrid.tsx

```diff
@@ -0,0 +1,269 @@
+// Alina TODO:
+//  - pagination and scroll
+// - submission
+
+import './DrugUploadTool.scss'
+
+import * as React from 'react'
+
+import { Link } from 'react-router-dom'
+
+import { Entity, } from '../../utils/jsonResponses/Entity'
+import { EntityLookupRequest } from '../../utils/jsonResponses/EntityLookupRequest'
+import { UserProfile } from '../../utils/jsonResponses/UserProfile'
+import { SynapseClient } from '../../utils'
+import { faTrash } from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import WarningModal from './WarningModal'
+
+
+import _ from 'lodash';
+import moment from 'moment';
+
+type UserData = {
+  fileList: Entity[];
+  existingData: any;
+  dataFolderId: string;
+}
+
+export type UserFileGridProps = {
+  token?: string;
+  parentContainerId: string; //projectId
+  pathpart: string;
+}
+
+type UserFileGridState = {
+  token?: string;
+  userprofile?: any;
+  dataFolderId?: string,
+  fileList: Entity[]
+  isLoading: boolean;
+  modalContext?: { 'action': Function, 'arguments': any[] }
+}
+
+
+export default class UserFileGrid extends React.Component<UserFileGridProps, UserFileGridState> {
+  modalTitle='Trash Submission?' 
+  modalCopy=<><p>This submission is currently incomplete and has not been submitted. If you trash this submission, you won't be able to recover the data.</p>
+  <p>Are you sure you want to trash this submission?' </p></>
+  
+  
+  constructor(props: UserFileGridProps) {
+    super(props)
+    this.state = {
+      isLoading: true,
+      token: this.props.token,
+      fileList: []
+    }
+  }
+
+  componentDidMount(): Promise<any> {
```

Comment:
As was already mentioned, token should come in as a prop and then componentDidMount would call refresh like it already is except it would pass in this.props.token, and then componentDidUpdate would check if the token has changed from the previous props and call refresh if the token has changed. And once that gets changed, I don't believe  componentDid(Mount/Update) and refresh should return a value?

Reply:
Returning the value is mostly for the sake of testing.
I think that way when we await for it we know that the promise it is calling has resolved?
I will just set it as async